Links with `:static` suffix do not generate correct target dependencies

**What seems to be the problem?**
Given this sample project where `links` is used with `:static` suffix to enable putting the static library in static section of link libraries (`-Wl,-Bstatic`):
```lua
workspace "MyWorkspace"
    configurations { "Debug" }
    toolset "gcc"

    project "MyLib"
        kind "StaticLib"
        files { "lib.cpp" }

    project "MyApp"
        kind "ConsoleApp"
        libdirs { "bin/Debug" }
        links { "MyLib:static" }
        files { "app.cpp" }
```
When premake is called to generate gmake Makefiles, and make is invoked to build MyApp:
```
$ premake5 gmake
$ make MyApp
```

Then it fails to build with the compiler not finding the MyLib:
```
==== Building MyApp (debug) ====
Creating obj/Debug/MyApp
app.cpp
Creating bin/Debug
Linking MyApp
/usr/bin/ld: cannot find -lMyLib
collect2: error: ld returned 1 exit status
make[1]: *** [MyApp.make:54: bin/Debug/MyApp] Error 1
make: *** [Makefile:31: MyApp] Error 2
```

This is because MyApp in generated Makefiles does not include a dependency on MyLib target. Therefore, MyLib is never built and the linking phase of MyApp fails.

**What did you expect to happen?**
Generated Makefiles should contain the correct dependency of MyApp on MyLib and `make MyApp` should build first MyLib, then MyApp successfully.

**What have you tried so far?**
The only workaround is to manually build MyLib first, then MyApp.

**How can we reproduce this?**
Run the example given above.

**What version of Premake are you using?**
premake5 (Premake Build Script Generator) 5.0.0-alpha16

**Anything else we should know?**
When not using the `:static` suffix, the dependencies are correct and the build works as expected:

However, that is not a valid workaround for the project I am working on. In this particular case, putting the library in `-Wl,-Bstatic` section is quite important for building the project correctly.

--- a/src/base/config.lua
+++ b/src/base/config.lua
@@ -262,10 +262,20 @@
 			local link = cfg.links[i]
 			local item
 
+			-- Strip linking decorators from link, to determine if the link
+			-- is a "sibling" project.
+			local endswith = function(s, ptrn)
+				return ptrn == string.sub(s, -string.len(ptrn))
+			end
+			local name = link
+			if endswith(name, ":static") or endswith(name, ":shared") then
+				name = string.sub(name, 0, -8)
+			end
+
 			-- Sort the links into "sibling" (is another project in this same
 			-- workspace) and "system" (is not part of this workspace) libraries.
 
-			local prj = p.workspace.findproject(cfg.workspace, link)
+			local prj = p.workspace.findproject(cfg.workspace, name)
 			if prj and kind ~= "system" then
 
 				-- Sibling; is there a matching configuration in this project that

--- a/tests/config/test_links.lua
+++ b/tests/config/test_links.lua
@@ -222,3 +222,45 @@
 		local r = prepare("all", "fullpath")
 		test.isequal({ "bin/Debug/MyProject2.lib" }, r)
 	end
+
+--
+-- Linking decorators need to be passed through
+--
+
+	function suite.canLink_StaticDecoratorSystemLib()
+		links { "SystemLibrary:static" }
+		local r = prepare("all", "fullpath")
+		test.isequal({ "SystemLibrary:static" }, r)
+	end
+
+	function suite.canLink_SharedDecoratorSystemLib()
+		links { "SystemLibrary:shared" }
+		local r = prepare("all", "fullpath")
+		test.isequal({ "SystemLibrary:shared" }, r)
+	end
+
+--
+-- Linking decorators need to be stripped for sibling projects
+--
+
+	function suite.canLink_StaticDecoratorSiblingLib()
+		links { "SiblingLibrary:static" }
+
+		project "SiblingLibrary"
+		kind "StaticLib"
+		language "C++"
+
+		local r = prepare("all", "fullpath")
+		test.isequal({ "bin/Debug/SiblingLibrary.lib" }, r)
+	end
+
+	function suite.canLink_SharedDecoratorSiblingLib()
+		links { "SiblingLibrary:shared" }
+
+		project "SiblingLibrary"
+		kind "SharedLib"
+		language "C++"
+
+		local r = prepare("all", "fullpath")
+		test.isequal({ "bin/Debug/SiblingLibrary.lib" }, r)
+	end

